grr/server/grr_response_server/gui/api_plugins/metadata_test.py

```diff
@@ -0,0 +1,382 @@
+        result_type = router_method.result_type is not None
+
+        # Check if there is a result type schema in the OpenAPI description.
+        # https://swagger.io/specification/#operation-object
+        operation_obj = \
+          self.openapi_desc_dict["paths"][simple_path][http_method]
+        responses_obj = operation_obj["responses"]
+
+        self.assertIn("200", responses_obj)
+        self.assertIn("default", responses_obj)
+
+        resp_200 = responses_obj["200"]
+        result_type_openapi = "content" in resp_200
+
+        self.assertEqual(result_type, result_type_openapi)
+
+  def testOpenApiPrimitiveTypesDescription(self):
+    operation_obj = (
+      self.openapi_desc_dict
+        .get("paths")
+        .get("/metadata_test/method3")
+        .get("get")
+    )
+    primitive_parameters = operation_obj["parameters"]
+
+    openapi_primitives = {p["name"]: p["schema"] for p in primitive_parameters}
+    openapi_primitives["BinaryStream"] = (
+      operation_obj
+        .get("responses")
+        .get("200")
+        .get("content")
+        .get("application/octet-stream")
+        .get("schema")
+    )
+
+    expected = dict()
+
+    expected_double = {"type": "number", "format": "double"}
+    expected["field_double"] = expected_double
+
+    expected_float = {"type": "number", "format": "float"}
+    expected["field_float"] = expected_float
+
+    expected_int64 = {"type": "integer", "format": "int64"}
+    expected["field_int64"] = expected_int64
+
+    expected_uint64 = {"type": "integer", "format": "uint64"}
+    expected["field_uint64"] = expected_uint64
+
+    expected_int32 = {"type": "integer", "format": "int32"}
+    expected["field_int32"] = expected_int32
+
+    expected_fixed64 = {"type": "integer", "format": "uint64"}
+    expected["field_fixed64"] = expected_fixed64
+
+    expected_fixed32 = {"type": "integer", "format": "uint32"}
+    expected["field_fixed32"] = expected_fixed32
+
+    expected_bool = {"type": "boolean"}
+    expected["field_bool"] = expected_bool
+
+    expected_string = {"type": "string"}
+    expected["field_string"] = expected_string
+
+    expected_bytes = {"type": "string", "format": "binary"}
+    expected["field_bytes"] = expected_bytes
+
+    expected_uint32 = {"type": "integer", "format": "uint32"}
+    expected["field_uint32"] = expected_uint32
+
+    expected_sfixed32 = {"type": "integer", "format": "int32"}
+    expected["field_sfixed32"] = expected_sfixed32
+
+    expected_sfixed64 = {"type": "integer", "format": "int64"}
+    expected["field_sfixed64"] = expected_sfixed64
+
+    expected_sint32 = {"type": "integer", "format": "int32"}
+    expected["field_sint32"] = expected_sint32
+
+    expected_sint64 = {"type": "integer", "format": "int64"}
+    expected["field_sint64"] = expected_sint64
+
+    expected_binarystream = {"type": "string", "format": "binary"}
+    expected["BinaryStream"] = expected_binarystream
```

Comment:
Would it be better to just build a Python object like:
```
expected = {
  ...
  "BinaryStream": {"type": "string", "format": "binary"}
}
```
?

Reply:
I agree. I made the changes to all such cases. Thank you!